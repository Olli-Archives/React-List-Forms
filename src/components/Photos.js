import React from 'react';
import Photo from './Photo';
import PropTypes from 'prop-types';

export default function Photos({ photos }) {
  const photoList = photos.map((photo, i) => {
    return (
      <li key={i}>
        <Photo photo={photo} />
      </li>
    );
  });
  return <ul>{photoList}</ul>;
}

Photos.propTypes = {
  photos: PropTypes.array.isRequired
}
;
