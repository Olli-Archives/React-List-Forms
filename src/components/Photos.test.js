import React from 'react';
import { shallow } from 'enzyme';
import Photos from './Photos';

const photosArray = ['https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg', 'https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg', 'https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg'];

describe('Photos component', ()=>{
  it('renders a list of photos from an array', ()=>{
    const wrapper = shallow(<Photos photos={photosArray}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
