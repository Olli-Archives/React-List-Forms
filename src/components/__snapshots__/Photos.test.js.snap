// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos component renders a list of photos from an array 1`] = `
Array [
  <li
    key="0"
  >
    <Photo
      photo="https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg"
    />
  </li>,
  <li
    key="1"
  >
    <Photo
      photo="https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg"
    />
  </li>,
  <li
    key="2"
  >
    <Photo
      photo="https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg"
    />
  </li>,
]
`;
